cxx4opencl/address_spaces.txt

```diff
@@ -229,6 +315,31 @@ __global const int& f(__global float &ref) {
 }
 ----------
 
-==== Initialization of local and constant address space objects
-
-TODO
+==== Construction, initialization and destruction
+
+Construction, initialization and destruction of objects in `__private`
+and `__global` address space follows the general principals of {cpp}. For
+the program scope objects the implementation (i.e. compiler) defines an ABI
+format for initialization and destruction of global objects before/after
+all kernels are enqueued.
+
+Objects in `__local` address space can not have initializers in
+declarations and therefore a constructor can not be called. All objects
+created in the local address space have undefined state at the beginning.
+Developers are free to defined a special member function that can
+initialize local address space objects after their declaration. Any
+default values provided for the initialization of members in a class
+declaration are ignored when creating the local address space objects.
+Classes that are instantiated with local address space objects can not
+have references as their members. The same restriction applies recursively
+to the members of such classes. Destructors of local address space objects
```

Comment:
I suppose this applies to base classes as well?

```suggestion
to the members and base classes. Destructors of local address space objects
```

Reply:
Absolutely! Thanks!